iml-system-test-utils/src/vagrant.rs

```diff
@@ -320,13 +321,57 @@ pub async fn setup_iml_install(
     Ok(())
 }
 
+pub async fn get_iml_devices(config: &ClusterConfig) -> Result<BTreeSet<String>, CmdError> {
+    let output = run_vm_command(config.manager, "iml devices list -d json")
+        .await?
+        .checked_output()
+        .await?;
+
+    let data_str = str::from_utf8(&output.stdout).expect("Couldn't parse devices information.");
+    let volumes: Vec<Volume> =
+        serde_json::from_str(data_str).expect("Couldn't serialize devices information.");
+
+    let labels: BTreeSet<String> = volumes.into_iter().map(|v| v.label).collect();
+
+    Ok(labels)
+}
+
+pub async fn wait_for_all_devices(max_tries: i32, config: &ClusterConfig) -> Result<(), CmdError> {
+    let mut count = 1;
+    let wwids: BTreeSet<String> = ssh::get_host_bindings(&config.storage_servers()[..]).await?;
+    println!("Comparing wwids from api to bindings: {:?}", wwids);
+
+    let iml_devices: BTreeSet<String> = get_iml_devices(config).await?;
+    let mut all_volumes_accounted_for = wwids.is_subset(&iml_devices);
+
+    println!("Comparing iml devices to bindings files.");
+    println!("iml_devices: {:?}", iml_devices);
+    println!("binding wwids: {:?}", wwids);
+
+    while !all_volumes_accounted_for && count < max_tries {
+        delay_for(Duration::from_secs(5)).await;
+
+        let iml_devices: BTreeSet<String> = get_iml_devices(config).await?;
+        all_volumes_accounted_for = wwids.is_subset(&iml_devices);
+        count += 1;
+
+        println!("Comparing iml devices to bindings files.");
+        println!("iml_devices: {:?}", iml_devices);
+        println!("binding wwids: {:?}", wwids);
+    }
```

Comment:
I am just pondering: :)
```rust
loop {
  count += 1;

  let iml_devices: BTreeSet<String> = get_iml_devices(config).await?;
  let all_volumes_accounted_for = wwids.is_subset(&iml_devices);
  
  println!("Comparing iml devices to bindings files ({})", &count);
  println!("iml_devices: {:?}", iml_devices);
  println!("binding wwids: {:?}", wwids);

  if all_volumes_accounted_for {
    return Ok();
  }
  if count >= max_tries {
    return Err(..);
  }
  delay_for(Duration::from_secs(5)).await;
}
```

Reply:
Yeah this could probably be updated. Looks a bit cleaner.